### PR TITLE
Baseimage 0.0.4

### PR DESCRIPTION
**Pull request by pSchlarb**
Baseimage changes as discussed in #1684

### CODE REVIEW

**Review by udosson — CHANGES_REQUESTED** [review]
@pSchlarb thank you very much for this PR.
This PR should only contain the updated Docker files and the shouldn't be referenced anywhere in the PR.
The workflow is the following.
- [x] Update Docker files (First PR - this PR)
- [x] Merge into master
- [ ] Build new Docker Images
- [ ] Push to DockerHub
- [ ] Update Build pipelines with new Docker image (`0.0.4`) (Second PR)
- [ ] Merge into master

**Comment by pSchlarb** [comment]
Will do and test the cython and linking part.

File: environment/docker/baseimage/indy-baseimage.ubuntu.dockerfile [hunk 1]
```diff
@@ -19,11 +19,14 @@ RUN apt-get update && apt-get install -y \
     python-setuptools
     
 # pypi based packages
-RUN pip3 install -U \ 
-    'pip<10.0.0' \
-    setuptools \
+RUN pip3 install  \
+    cython\
```
[the remark this answers is not in the document]
Comment (pSchlarb) [reply environment/docker/baseimage/indy-baseimage.ubuntu.dockerfile:23]: If it is not installed the pip install .[tests] complains about  `distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('Cython>=0.20')`

File: environment/docker/baseimage/indy-baseimage.ubuntu.dockerfile [hunk 2]
```diff
@@ -19,11 +19,14 @@ RUN apt-get update && apt-get install -y \
...
     virtualenv
 
+RUN pip3 install -U "setuptools<=50.3.2" 
+
+RUN ln -s /usr/bin/pip3 /usr/bin/pip
```
[the remark this answers is not in the document]
Comment (pSchlarb) [reply environment/docker/baseimage/indy-baseimage.ubuntu.dockerfile:28]: Yeah it works. The `pip install .[tests]` and the test are running fine.

**Comment by pSchlarb** [comment]
@udosson I will update #1684 after this PR is merged, if its okay?

File: environment/docker/baseimage/indy-baseimage.ubuntu.dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (udosson) [reply environment/docker/baseimage/indy-baseimage.ubuntu.dockerfile:23]: Could you replace your `pip3 install` with the following, please?
```
RUN pip3 install -U \
    pip==20.3.4 \
    virtualenv==20.0.35 \
    "setuptools<=50.3.2"
```

**Review by udosson — CHANGES_REQUESTED** [review]
@pSchlarb thanks for applying the requested changes. There is only one thing to do.
When building your image, `pip` is at version `8.1.1`.
```
$ pip list
pip (8.1.1)
```

I recommend installing the latest supported version of pip which is at `20.0.35`.
`cython` shouldn't be needed then. 
Could you very please if `pip install .[tests]` works with the new image on your PR #1684?

**Comment by pSchlarb** [comment]
With that newer pip version you have to use `--use-deprecated legacy-resolver` or pip will complain about not finding the correct plenum packages:
```
  WARNING: Requested python3-indy==1.15.0-dev-1625 from https://files.pythonhosted.org/packages/a0/78/74c1d6206c1eae93f677abd6ff03bd6357412716615f0df6858bd655b6a1/python3-indy-1.15.0-dev-1625.tar.gz#sha256=821729e796a47fd591520c8c73fefc91dce7bfc9f784c6702ca9d03d82f2e76a (from indy-node==1.13.0.dev0), but installing version 1.15.0
WARNING: Discarding https://files.pythonhosted.org/packages/a0/78/74c1d6206c1eae93f677abd6ff03bd6357412716615f0df6858bd655b6a1/python3-indy-1.15.0-dev-1625.tar.gz#sha256=821729e796a47fd591520c8c73fefc91dce7bfc9f784c6702ca9d03d82f2e76a (from https://pypi.org/simple/python3-indy/). Requested python3-indy==1.15.0-dev-1625 from https://files.pythonhosted.org/packages/a0/78/74c1d6206c1eae93f677abd6ff03bd6357412716615f0df6858bd655b6a1/python3-indy-1.15.0-dev-1625.tar.gz#sha256=821729e796a47fd591520c8c73fefc91dce7bfc9f784c6702ca9d03d82f2e76a (from indy-node==1.13.0.dev0) has different version in metadata: '1.15.0'
ERROR: Could not find a version that satisfies the requirement python3-indy==1.15.0-dev-1625 (from indy-node[tests])
ERROR: No matching distribution found for python3-indy==1.15.0-dev-1625
```
And thereafter complains about zipp, which seems to be a dependency of `importlib-metadata>=2.0` comming from plenum:
```
Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.5/dist-packages (from importlib-metadata>=2.0->indy-plenum==1.13.0.dev143->indy-node==1.13.0.dev0) (3.6.0)
ERROR: Package 'zipp' requires a different Python: 3.5.2 not in '>=3.6'
```

**Comment by pSchlarb** [comment]
@udosson Should `zipp` be therefore pinned in the baseimages, since the requirement for it is only `>=0.5`?

**Comment by udosson** [comment]
How could we solve the issue with the newer version of pip without using `--use-deprecated legacy-resolver`? Do you know the root of the issue?

`zipp`
The error message complains about a version of `zipp` that has no support for Python 3.5. I pinned `importlib-metadata`to `v2.1.1`in this commit (https://github.com/hyperledger/indy-plenum/commit/5bef312984622f9a0d2501c3e0fee6de2dd283ac) 
You could therefore use another version of Indy-Plenum in `setup.py` which should resolve the issue. Namely the PyPI packages that gets published in this gha run https://github.com/hyperledger/indy-plenum/actions/runs/1418274109

**Comment by pSchlarb** [comment]
To the issue of the `deprecated legacy-resolver`:
In the old versions of pip it doesn't check the versions of packages strictly.
And the versions mismatch with the metadata provided. (If i recall it correctly in the metadata gotten from pypi only is something like `1.15.0`is set but the setup.py requires `python3-indy==1.15.0-dev-1625`)
I will check where exactly the issue is.

**Comment by pSchlarb** [comment]
I think i found the issue with `python-indy` and deprecated resolvers:
In https://github.com/hyperledger/indy-sdk/blob/master/wrappers/python/setup.py line 4:
`PKG_VERSION = os.environ.get('PACKAGE_VERSION') or '1.15.0'`
the version is statically set to `1.15.0` if the `PACKAGE_VERSION` file doesn't exist.

**Comment by udosson** [comment]
@pSchlarb thanks for adjusting the PR. Could you squash merge your commits into one, please?
Then I'll merge your PR.

**Comment by pSchlarb** [comment]
Yes zipp is installed and the venv is working.
And the tests are running properly.

**Comment by udosson** [comment]
(ci) test this please

**Review by udosson — APPROVED** [review]
LGTM. Thanks @pSchlarb

**Pull request merged**